service/src/test/java/com/sedliarov/learningtable/integration/StudentControllerIntegrationTests.java

```diff
@@ -0,0 +1,202 @@
+
+    // when
+    ResponseEntity<StudentDto[]> response = exchangeGetWithoutAuth(STUDENTS_URL, StudentDto[].class);
+
+    // then
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(response.getBody()).containsExactlyElementsOf(expectedStudents);
+  }
+
+  @Test
+  void negativeTestGetStudents() {
+
+    // given
+    List<StudentDto> expectedStudents = new ArrayList<>();
+
+    // when
+    ResponseEntity<StudentDto[]> response = exchangeGetWithoutAuth(STUDENTS_URL, StudentDto[].class);
+
+    // then
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(response.getBody()).containsExactlyElementsOf(expectedStudents);
+  }
+
+  @Test
+  void testCreateStudent() {
+
+    // given
+    StudentDto newStudent = StudentFixture.createDto();
+
+    // when
+    ResponseEntity<StudentDto> student =
+        exchangeAddWithoutAuth(STUDENTS_URL, newStudent, StudentDto.class);
+
+    // then
+    assertThat(student.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+    assertThat(student.getBody().getFirstName()).isEqualTo(newStudent.getFirstName());
+    assertThat(student.getBody().getSecondName()).isEqualTo(newStudent.getSecondName());
+    assertThat(student.getBody().getNote()).isEqualTo(newStudent.getNote());
+    assertThat(student.getBody().getGroup()).isEqualTo(newStudent.getGroup());
+  }
+
+  @Test
+  void negativeTestCreateStudent() {
+
+    // given
+    StudentDto newStudent = StudentFixture.createDto();
+
+    // when
+    ResponseEntity<StudentDto> student =
+        exchangeAddWithoutAuth(STUDENTS_URL + "/" + newStudent.getStudentId(), newStudent, StudentDto.class);
+
+    // then
+    assertThat(student.getStatusCode()).isEqualTo(HttpStatus.METHOD_NOT_ALLOWED);
+  }
+
+  @Test
+  void testDeleteStudent() {
+
+    // given
+    Student newStudent = StudentFixture.createEntity();
+    studentRepository.save(newStudent);
+
+    // when
+    ResponseEntity<StudentDto> student =
+        exchangeDeleteWithoutAuth(STUDENTS_URL + "/" + newStudent.getStudentId(), StudentDto.class);
+
+    // then
+    ResponseEntity<StudentDto> check =
+        exchangeGetWithoutAuth(STUDENTS_URL + "/" + newStudent.getStudentId(), StudentDto.class);
+    assertThat(student.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(check.getStatusCode()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  void negativeTestDeleteStudent() {
+
+    // given
+    studentRepository.save(StudentFixture.createEntity());
+
+    // when
+    ResponseEntity<StudentDto> student =
+        exchangeDeleteWithoutAuth(STUDENTS_URL, StudentDto.class);
+
+    // then
+    ResponseEntity<StudentDto[]> response = exchangeGetWithoutAuth(STUDENTS_URL, StudentDto[].class);
+    List<StudentDto> students = Arrays.stream(response.getBody()).toList();
+    assertThat(student.getStatusCode()).isEqualTo(HttpStatus.METHOD_NOT_ALLOWED);
+    assertThat(students).isNotEmpty();
+  }
```

Comment:
Name of test and also not correct flow. Pls check needed example or add needed flow

Reply:
Done